crates/datasources/src/sqlite/wrapper.rs

```diff
@@ -0,0 +1,184 @@
+use std::fmt;
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use async_sqlite::rusqlite::types::Value;
+use async_sqlite::rusqlite::{self, OpenFlags};
+use futures::Stream;
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+
+use crate::sqlite::errors::{Result, SqliteError};
+
+#[derive(Clone)]
+pub struct SqliteAsyncClient {
+    path: PathBuf,
+    inner: async_sqlite::Client,
+}
+
+impl fmt::Debug for SqliteAsyncClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "SqliteAsyncClient({})", self.path.to_string_lossy())
+    }
+}
+
+impl SqliteAsyncClient {
+    pub async fn new(path: PathBuf) -> Result<Self> {
+        let inner = async_sqlite::ClientBuilder::new()
+            .flags(OpenFlags::SQLITE_OPEN_READ_ONLY)
+            .path(&path)
+            .open()
+            .await?;
+        Ok(Self { path, inner })
+    }
+
+    pub fn query(&self, s: impl Into<String>) -> SqliteQueryStream {
+        let s = s.into();
+
+        let (tx, rx) = mpsc::channel(1);
+
+        let client = self.inner.clone();
+        let handle = tokio::spawn(async move {
```

Comment:
Can't really do this because the future is never polled (`.await`ed), i.e., the set of instructions inside `async {}` block are never executed. `spawn`ing here is necessary since the future inside spawn is polled at-least once.

We could go around it by polling the future once but it just makes the code way more complicated and brings nothing to value here, also, we never poll it again and since rust's futures are lazy, we can have a dangling future in memory. Moreover, spawning here makes sense.

Reply:
My thought was that this method returns a stream, which means that we don't need to work until that stream is polled, so it's fine that it's lazy, and in fact maybe preferable if we don't do any work until the stream is polled. (though this gets weird for the error handling. 

If you have a `futures::stream::Once` that returns a `Stream` object, you can call `flatten()` on that stream object (via `StreamExt`) and then you have it... no background threads. (I think)

you do need the background thread if you want to collect _all_ of the results without the stream, but you could also just iterate the stream and save the code.

Reply:
Did it without spawning. Did not use stream but should do it!